src/xrpld/app/misc/NetworkOPs.cpp

```diff
@@ -1509,13 +1529,17 @@ NetworkOPsImp::apply(std::unique_lock<std::mutex>& batchLock)
             {
                 auto const toSkip =
                     app_.getHashRouter().shouldRelay(e.transaction->getID());
-
-                if (toSkip)
+                if (auto const sttx = *(e.transaction->getSTransaction());
+                    toSkip &&
+                    // Skip relaying if it's an inner batch txn and batch
+                    // feature is enabled
+                    !(sttx.isFlag(tfInnerBatchTxn) &&
+                      newOL->rules().enabled(featureBatch)))
```

Comment:
is the check `newOL->rules().enabled(featureBatch)` necessary? IMO, If inner batch flag is set, we want always want to skip it.

Hypothetically, assume there is a transaction with `tfInnerBatchTxn` flag but `featureBatch` is disabled, then this transaction would get relayed.

Reply:
I think it could be removed but it was suggested to be added by @mvadari so I just added it. Yes, when batch is not enabled it will relay those transactions, however, because its also not enabled, it would not skip the signature check. So it would get rejected later.